builder/builder.go

```diff
@@ -0,0 +1,210 @@
+	}
+
+	hostData.Environment = configuration.Envtype
+	hostData.Location = configuration.Location
+	hostData.HostType = configuration.HostType
+	hostData.Version = version
+	hostData.HostDataSchemaVersion = hostDataSchemaVersion
+	hostData.Info = *getHost(configuration)
+	hostData.Hostname = hostData.Info.Hostname
+
+	hostData.Extra.Filesystems = getFilesystems(configuration)
+	hostData.Extra.Databases = getDatabases(configuration, hostData.Info.Type)
+
+	hostData.Databases, hostData.Schemas = getDatabasesAndSchemaNames(hostData.Extra.Databases)
+
+	return hostData
+}
+
+func getHost(configuration config.Configuration) *model.Host {
+	out := fetcher(configuration, "host")
+	host := marshal.Host(out)
+
+	host.Environment = configuration.Envtype
+	host.Location = configuration.Location
+
+	return &host
+}
+
+func getFilesystems(configuration config.Configuration) []model.Filesystem {
+	out := fetcher(configuration, "filesystem")
+	return marshal.Filesystems(out)
+}
+
+func getDatabases(configuration config.Configuration, hostType string) []model.Database {
+	out := fetcher(configuration, "oratab", configuration.Oratab)
+	oratabEntries := marshal.Oratab(out)
+
+	databaseChannel := make(chan model.Database, len(oratabEntries))
+
+	for _, entry := range oratabEntries {
+		utils.RunRoutine(configuration, func() {
+			databaseChannel <- getDatabase(configuration, entry, hostType)
+		})
+	}
+
+	var databases = []model.Database{}
+	for i := 0; i < len(oratabEntries); i++ {
+		databases = append(databases, <-databaseChannel)
+	}
+
+	return databases
+}
+
+func getDatabase(configuration config.Configuration, entry model.OratabEntry, hostType string) model.Database {
+	dbStatusOut := fetcher(configuration, "dbstatus", entry.DBName, entry.OracleHome)
+	dbStatus := strings.TrimSpace(string(dbStatusOut))
+
+	var database model.Database
+
+	switch dbStatus {
+	case "OPEN":
+		database = getOpenDatabase(configuration, entry, hostType)
+	case "MOUNTED":
+		{
+			out := fetcher(configuration, "dbmounted", entry.DBName, entry.OracleHome)
+			database = marshal.Database(out)
+
+			database.Tablespaces = []model.Tablespace{}
+			database.Schemas = []model.Schema{}
+			database.Patches = []model.Patch{}
+			database.Features = []model.Feature{}
+			database.Licenses = []model.License{}
+			database.ADDMs = []model.Addm{}
+			database.SegmentAdvisors = []model.SegmentAdvisor{}
+			database.LastPSUs = []model.PSU{}
+			database.Backups = []model.Backup{}
+		}
+	default:
+		log.Println("Error! DBName: " + entry.DBName + "OracleHome: " + entry.OracleHome + "  Wrong dbStatus: " + dbStatus)
+	}
+
+	return database
+}
+
+func getOpenDatabase(configuration config.Configuration, entry model.OratabEntry, hostType string) model.Database {
+	dbVersionOut := fetcher(configuration, "dbversion", entry.DBName, entry.OracleHome)
+	dbVersion := strings.Split(string(dbVersionOut), ".")[0]
+
+	statsCtx, cancelStatsCtx := context.WithCancel(context.Background())
```

Comment:
Are you sure that context.Background is correct?
[godoc](https://golang.org/pkg/context/#TODO) says that it's never cancelled

Reply:
context.Background() is the parent context, WithCancel returns a copy of parent with a new Done channel
https://golang.org/pkg/context/#WithCancel